libcloudforensics/providers/gcp/internal/build.py

```diff
@@ -89,17 +90,34 @@ def BlockOperation(self, response: Dict[str, Any]) -> Dict[str, Any]:
           operations.
 
     Raises:
-      RuntimeError: If API call failed.
+      RuntimeError: If API call failed or if getting the Cloud Build
+          API operation object failed.
     """
     service = self.GcbApi()
     while True:
       request = service.operations().get(name=response['name'])
-      response = request.execute()
+      get_success = False
+      for block_retry in range(BLOCK_RETRY_MAX):
+        try:
+          response = request.execute()
+          get_success = True
+        except googleapiclient.errors.HttpError as error:
+          logging.info(
+              'build.BlockOperation: Get request to cloudbuild.googleapis.com '
+              'failed.\nTry {0:d} of {1:d}. Error: {2!s} '.format(
+                  block_retry, BLOCK_RETRY_MAX, error))
+        if get_success:
+          break
+        if block_retry == BLOCK_RETRY_MAX - 1:
+          raise RuntimeError(
+              'Faliure blocking Cloud Build operation: {0:s}'.format(
+                  response['name']))
```

Comment:
This function is pretty complex and has two levels of retries, one where we catch HTTP errors, limited to a MAX number and the other where we catch other types of errors, not capped (so will retry until a response is obtained.

I think it would make sense in this case to have the retry loops in different functions (just add a new one to do the capped http exception catching, something like _RetryExecuteRequest()). No need to declare the BLOCK_RETRY_MAX as a module-level variable since it's only used by this function. Also consider using the `for / else` construct which might be of use here.

Reply:
Done